tests/functional/test_rose_stem.py

```diff
@@ -280,65 +279,63 @@ def _inner_fn(rose_stem_opts, verbosity=verbosity):
     yield _inner_fn
 
 
-@pytest.fixture(scope='class')
-def rose_stem_run_really_basic(rose_stem_run_template, setup_stem_repo):
+@pytest.fixture()
+async def rose_stem_run_really_basic(rose_stem_run_template, setup_stem_repo):
     rose_stem_opts = {
         'stem_groups': [],
         'stem_sources': [
             str(setup_stem_repo['workingcopy']), "fcm:foo.x_tr@head"
         ],
     }
-    yield rose_stem_run_template(rose_stem_opts)
+    yield await rose_stem_run_template(rose_stem_opts)
 
 
-class TestReallyBasic():
-    def test_really_basic(self, rose_stem_run_really_basic):
-        """Check that assorted variables have been exported.
-        """
-        assert rose_stem_run_really_basic['run_stem'].returncode == 0
+def test_really_basic(rose_stem_run_really_basic):
+    """Check that assorted variables have been exported.
+    """
+    assert rose_stem_run_really_basic['run_stem'].returncode == 0
 
 
-@pytest.fixture(scope='class')
-def rose_stem_run_basic(rose_stem_run_template, setup_stem_repo):
+@pytest.fixture()
+async def rose_stem_run_basic(rose_stem_run_template, setup_stem_repo):
     rose_stem_opts = {
         'stem_groups': ['earl_grey', 'milk,sugar', 'spoon,cup,milk'],
         'stem_sources': [
             str(setup_stem_repo['workingcopy']), "fcm:foo.x_tr@head"
         ],
         'workflow_name': setup_stem_repo['suitename']
     }
-    yield rose_stem_run_template(rose_stem_opts)
-
-
-class TestBasic():
```

Comment:
These test classes aren't achieving anything since they each contain only one test and don't manage any resources.

Test classes date back to `unittest`, fixtures are more modern solution. Working with module-scoped fixtures and class-scoped tests gets tricky.